src/katsdpmodels/fetch.py

```diff
@@ -263,11 +264,12 @@ def resolve(self, url: str) -> List[str]:
             parts = urllib.parse.urlparse(url)
         return chain
 
-    def _get_eager(self, url: str) -> Tuple[io.IOBase, str, str]:
+    def _get_eager(self, url: str) -> Tuple[io.IOBase, str, Optional[str], str]:
```

Comment:
I don't really see how the positional content_type is optional ?

Reply:
It can be None e.g. if the HTTP header is not present.